### PR TITLE
Update integration lib packages & cache reset api

### PR DESCRIPTION
**Pull request by HansDahle**
- [x] New feature
- [ ] Bug fix
- [x] High impact

**Description of work:**
Was only going to add a simple endpoint to trigger cache reset across deployed instances. 
Utilizing distributed mediatr lib to coordinate across instances. 

This caused a larger nuget upgrade scenario, so the main workload here is related to the nuget upgrade. 
This also solves the discovery service issue, which promotes integration lib upgrade.

MediatR lib needed to be upgraded which had breaking changes, deprecated the AsyncRequestHandler abstract type, requiring switch to IRequestHandler. 
This change also impacted how requests without responses were handled. Behaviours needed updates as well as generic methods which cause ambigous signature issues (DispatchCommand).

**Testing:**
- [x] Can be tested
- [x] Automatic tests created / updated
- [x] Local tests are passing

Needed to update fixtures due to changes in MediatR setup logic and introduction of new background services and integration which needed new mocks. Distributed mediatr have service bus integration which requires connection string. This is mocked with services doing nothing (if we need to verify distributed message handling a new mock will need to be created).

**Checklist:**
- [x] Considered automated tests
- [ ] Considered updating specification / documentation
- [ ] Considered work items 
- [x] Considered security
- [x] Performed developer testing
- [x] Checklist finalized / ready for review

Should verify cache reset endpoint, to verify that distributed infra is set up correctly.

### CODE REVIEW

File: src/backend/api/Fusion.Resources.Api/Controllers/AdminController.cs [hunk 1]
```diff
@@ -12,22 +16,69 @@ public class AdminController : ResourceControllerBase
...
+        public AdminController(IOrgUnitCache orgUnitCache, IMediator mediator)
         {
 
             this.orgUnitCache = orgUnitCache;
+            this.mediator = mediator;
         }
 
         [HttpGet("admin/cache/org-units")]
 
         public async Task<ActionResult> CleareCache()
```
Comment (terjebra) [inline src/backend/api/Fusion.Resources.Api/Controllers/AdminController.cs:30]: Small typo there :)

**Comment by HansDahle** [comment]
Verified cache reset endpoint.

```
POST https://resources-api-pr-597.fusion-dev.net/admin/cache/reset-internal-cache
{
    "message": "Cache reset has been queued for all instances."
}
```

```
❯ fdev sb listen sb-fap-resources-pr -t dist-notifications
Found connection string to sb-fap-resources-pr
Created subscription fdev-04a14e19-6184-4c44-8ddb-f552f888c2d8 on topic dist-notifications
Registered receiver on topic dist-notifications
Listening...
Press enter to exit
---- New message @ 16/10/2023 12:11:19
-- Message Id: 05e96bd9983042d3be1ba526091a129a
-- Content-Type: application/json
-- ApplicationProperty: crlType=Fusion.Resources.Api.DistributedEvents.ResetCacheNotification
-- ApplicationProperty: sourceId=6282d13f-a5b9-4c88-ad03-d8a0e30e5aea
-- ApplicationProperty: scope=Domain
-- ApplicationProperty: sourceService=Fusion.Resources.Api, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
-- ApplicationProperty: Diagnostic-Id=00-92c58f8dcb7bb87ff13823f25edef648-03429d086f5a9aef-00
-- Payload:
{
  "Context": {
    "NotificationId": "c163bfc4-f192-4df7-aa6f-65966b071eeb",
    "SourceId": "6282d13f-a5b9-4c88-ad03-d8a0e30e5aea",
    "IsLocal": true,
    "Scope": 0
  }
}
```

**Review by terjebra — APPROVED** [review]

File: src/backend/api/Fusion.Resources.Api/Controllers/AdminController.cs (hunk 1 above)
Comment (HansDahle) [reply src/backend/api/Fusion.Resources.Api/Controllers/AdminController.cs:30]: IMediator mediator, can't see it 🙈 or do you mean it should be mediatr without the o?

**Comment by HansDahle** [comment]
Verified pr function app is triggering cache reset endpoint. Updated function app pr deployment pipeline so it is targeting the api correctly.

**Pull request merged**